ntoskrnl/mm/section.c

```diff
@@ -3029,9 +3035,12 @@ MmCreateDataFileSection(PROS_SECTION_OBJECT *SectionObject,
             Segment->RawLength.QuadPart = MaximumSize.QuadPart;
             Segment->Length.QuadPart = PAGE_ROUND_UP(Segment->RawLength.QuadPart);
         }
+
+        /* We let the segment reference the file object */
+        ObDereferenceObject(FileObject);
```

Comment:
The fact that this function takes ownership of the caller's reference is super unintuitive. Would be nicer if we instead called ObReferenceObject in the case where we store the passed-in pointer, and let the caller do its own deref. Same for `MmCreateImageSection`, obviously.

Reply:
True. And there was a leak, also. Fixed.